src/middlewared/middlewared/plugins/datastore/write.py

```diff
@@ -29,7 +28,11 @@ class Config:
                     insert.setdefault(column.name, '')
 
         await self.middleware.call('datastore.execute_write', table.insert().values(**insert))
-        pk = (await self.middleware.call('datastore.fetchall', 'SELECT last_insert_rowid()'))[0][0]
+        row_id = (await self.middleware.call('datastore.fetchall', 'SELECT last_insert_rowid()'))[0][0]
+        pk = (await self.middleware.call(
```

Comment:
We can probably avoid extra query. If primary key is integer, we return rowid. If primary key is not integer, we already have it in `insert`. Also, please add tests.

Reply:
That would be generally true ( at least right now it is I think ) but if say we were to have a table which has an integer as a primary key which is not `id` ?

Reply:
In SQLite integer primary key will automatically be an alias for rowid